ci-operator/step-registry/cucushift/installer/check/gcp/credentials-mode/cucushift-installer-check-gcp-credentials-mode-commands.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+
+GOOGLE_PROJECT_ID="$(< ${CLUSTER_PROFILE_DIR}/openshift_gcp_project)"
+export GCP_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/gce.json"
+sa_email=$(jq -r .client_email ${GCP_SHARED_CREDENTIALS_FILE})
+if ! gcloud auth list | grep -E "\*\s+${sa_email}"
+then
+  gcloud auth activate-service-account --key-file="${GCP_SHARED_CREDENTIALS_FILE}"
+  gcloud config set project "${GOOGLE_PROJECT_ID}"
+fi
+
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+    # shellcheck disable=SC1090
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+ret=0
+
+echo "INFO: Checking cluster credentials mode..."
+
+curr_credentials_mode=$(oc get cloudcredential cluster -o json | jq -r ".spec.credentialsMode")
+if [ -z "${curr_credentials_mode}" ]; then
+    curr_credentials_mode="Mint"
+fi
+echo "INFO: Current credentials mode: '${curr_credentials_mode}'"
+
+expected_credentials_mode=""
+if [ -n "${CREDENTIALS_MODE}" ]; then
+    expected_credentials_mode="${CREDENTIALS_MODE}"
+elif test -f "${SHARED_DIR}/install-config.yaml"; then
+    expected_credentials_mode=$(yq-go r "${SHARED_DIR}/install-config.yaml" credentialsMode)
+fi
+if [ -z "${expected_credentials_mode}" ]; then
+    expected_credentials_mode="${curr_credentials_mode}"
+fi
+echo "INFO: The expected credentials mode: '${expected_credentials_mode}'"
+
+if [[ "${curr_credentials_mode}" == "${expected_credentials_mode}" ]]; then
+    echo "INFO: Cluster credentials mode check passed."
+else
+    echo "ERROR: Cluster credentials mode check failed."
+    ret=$(( $ret | 1 ))
+fi
+
+if [[ "${curr_credentials_mode}" == "Passthrough" ]]; then
+    echo "INFO: Cluster credentials mode is 'Passthrough', checking the IAM service-accounts..."
+    echo "INFO: Expecting only one IAM service-account, and it is for worker nodes."
+    readarray -t iam_accounts < <(gcloud iam service-accounts list --filter="displayName~${CLUSTER_NAME}" --format="table(email)" | grep -v EMAIL)
+    count=0; found=false
+    for iam_account in "${iam_accounts[@]}";
+    do
+        count=$(( $count + 1 ))
+        if [[ "${iam_account}" =~ \-w\@.+\.iam\.gserviceaccount\.com ]]; then
```

Comment:
Per your test result:
```
INFO: Cluster credentials mode is 'Passthrough', checking the IAM service-accounts...
INFO: Expecting only one IAM service-account, and it is for worker nodes.
INFO: 1 - 'ci-op-wpf3dz95-49ce9-96h2d-w@XXXXXXXXXXXX.iam.gserviceaccount.com' (true)
```
I think we can get infra id from `${SHARED_DIR}/metadata.json` to expect `<infra-id>\-w\@.+\.iam\.gserviceaccount\.com` to be matched.